tensorpipe/test/channel/channel_test.h

```diff
@@ -202,6 +203,62 @@ class ChannelTestCase {
 
 template <typename TBuffer>
 class ClientServerChannelTestCase : public ChannelTestCase<TBuffer> {
+  std::vector<std::future<std::shared_ptr<tensorpipe::transport::Connection>>>
+  accept(
+      std::shared_ptr<tensorpipe::transport::Listener> listener,
+      size_t numConnections) {
+    auto promises = std::make_shared<std::vector<
+        std::promise<std::shared_ptr<tensorpipe::transport::Connection>>>>(
+        numConnections);
+    for (size_t i = 0; i < numConnections; ++i) {
+      listener->accept(
+          [&, promises](
+              const tensorpipe::Error& error,
+              std::shared_ptr<tensorpipe::transport::Connection> connection) {
+            ASSERT_FALSE(error) << error.what();
```

Comment:
Should we set the corresponding promise to error in this case, so that the main thread doesn't get blocked forever?

Reply:
Making the promise error would just delay the throwing of the exception, no? What did you have in mind?

Reply:
Would it? My worry is that on line 284 we call `get()` on the future, and if we throw the error here then this thread crashes (which could be some event loop), and thus the future will never be completed, and thus the main thread on like 284 will hang forever.